plonky2/src/gadgets/nonnative.rs

```diff
@@ -302,6 +633,41 @@ mod tests {
         verify(proof, &data.verifier_only, &data.common)
     }
 
+    fn test_nonnative_many_muls_helper(num: usize) {
+        type FF = Secp256K1Base;
+        const D: usize = 2;
+        type C = PoseidonGoldilocksConfig;
+        type F = <C as GenericConfig<D>>::F;
+
+        let config = CircuitConfig::standard_ecc_config();
+        let mut unop_builder = CircuitBuilder::<F, D>::new(config.clone());
+        let mut op_builder = CircuitBuilder::<F, D>::new(config);
+
+        let ffs: Vec<_> = (0..num).map(|_| FF::rand()).collect();
+
+        let op_targets: Vec<_> = ffs
+            .iter()
+            .map(|&x| op_builder.constant_nonnative(x))
+            .collect();
+        op_builder.mul_many_nonnative(&op_targets);
+
+        let unop_targets: Vec<_> = ffs
+            .iter()
+            .map(|&x| unop_builder.constant_nonnative(x))
+            .collect();
+        let mut result = unop_targets[0].clone();
+        for i in 1..unop_targets.len() {
+            result = unop_builder.mul_nonnative(&result, &unop_targets[i]);
+        }
+    }
```

Comment:
What does this do? It seems like it uses 2 `CircuitBuilder`s, but no proofs are created.

Reply:
Oh hmm — I think I just forgot to finish it. The idea is to compare calling `mul_nonnative` several times, versus one call to `mul_many_nonnative`... but I don't think that matters since "mul_many" isn't even optimized

Reply:
I'll just get rid of the whole test